source/common/network/utility.cc

```diff
@@ -606,16 +606,18 @@ void passPayloadToProcessor(uint64_t bytes_read, Buffer::InstancePtr buffer,
 Api::IoCallUint64Result readFromSocketRecvGro(IoHandle& handle,
                                               const Address::Instance& local_address,
                                               UdpPacketProcessor& udp_packet_processor,
-                                              MonotonicTime receive_time,
-                                              uint32_t* packets_dropped) {
+                                              MonotonicTime receive_time, uint32_t* packets_dropped,
+                                              uint32_t* num_packets_read) {
   ASSERT(Api::OsSysCallsSingleton::get().supportsUdpGro(),
          "cannot use GRO when the platform doesn't support it.");
   Buffer::InstancePtr buffer = std::make_unique<Buffer::OwnedImpl>();
   IoHandle::RecvMsgOutput output(1, packets_dropped);
 
-  // TODO(yugant): Avoid allocating 24k for each read by getting memory from UdpPacketProcessor
+  // TODO(yugant): Avoid allocating 64k for each read by getting memory from UdpPacketProcessor
   const uint64_t max_rx_datagram_size_with_gro =
-      NUM_DATAGRAMS_PER_RECEIVE * udp_packet_processor.maxDatagramSize();
+      num_packets_read != nullptr
+          ? 64 * 1024
```

Comment:
I'm also debating if we want to flag protect this at all. Apparently the old buffer size makes GRO unusable on client side. But I'm not sure if anyone is using GRO on server side which is default off. On server side, GRO might not have the MSG_TRUNC issue because likely the server won't continuously read more than 24kB payload from one client.

If we want to flag protect this, latching it with the read limit change makes sense to me because recvmsg should use a bigger buffer only if it knows the caller is counting packets read.

Reply:
> But I'm not sure if anyone is using GRO on server side which is default off

That's true, some users could be turning GRO on server side.  I think we can keep it as is then.